autoreject/autoreject.py

```diff
@@ -830,6 +849,22 @@ def __repr__(self):
         return '%s(%s)' % (class_name, _pprint(params,
                            offset=len(class_name),),)
 
+    def __getstate__(self):
```

Comment:
One more question: what do we buy into by using a '__' method? Why not just use a regular private method like _as_dict or so?

Reply:
I was just copying the code @wmvanvliet used for writing the report. I am not 100% sure what his rationale was but I think this will ensure that you get the same object weather you `pickle` or use `hdf5`

Reply:
Ok cool. Just let's make sure it's not random fancy stuff but specifically purposeful.

Reply:
The __getstate__ method is for returning a serializable dictionary. As far as I know, it is only used by pickle, cloudpickle and the like. But just like using __len__  for returning the length, why not always use a dunder method if there is an appropriate one? Isn't that more pythonic?

Reply:
I see h5io as in the same family as pickle and friends, so for me it makes sense to use __getstate__

Reply:
+1